storage/src/cache/dedup/db.rs

```diff
@@ -0,0 +1,329 @@
+// Copyright (C) 2022-2023 Alibaba Cloud. All rights reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#![allow(unused)]
+
+use std::path::Path;
+
+use r2d2::{Pool, PooledConnection};
+use r2d2_sqlite::SqliteConnectionManager;
+use rusqlite::{Connection, DropBehavior, OptionalExtension, Transaction};
+
+use super::Result;
+
+pub struct CasDb {
+    pool: Pool<SqliteConnectionManager>,
+}
+
+impl CasDb {
+    pub fn new(path: impl AsRef<Path>) -> Result<CasDb> {
+        let mut db_path = path.as_ref().to_owned();
+        db_path.push("cas.db");
+        Self::from_file(db_path)
+    }
+
+    pub fn from_file(db_path: impl AsRef<Path>) -> Result<CasDb> {
+        let mgr = SqliteConnectionManager::file(db_path);
+        let pool = r2d2::Pool::new(mgr)?;
+        let conn = pool.get()?;
+
+        conn.execute(
+            "CREATE TABLE IF NOT EXISTS Blobs (
+            BlobId  INTEGER PRIMARY KEY,
```

Comment:
Not sure if we need to consider fs_version, digester, chunk_size, compressor etc.

Reply:
The `ChunkId` contains digest algorithm prefixes, such as `sha256:xxx`.
And it only records chunks in uncompressed plaintext, so no need for `compressor` and `fs_version`.
For `chunk_size`, I'm a little hesitating whether to include it.

Reply:
Do we have any way of versioning or invalidating the chunk db and how to recycle chunks/blobs that are no longer needed.

Reply:
This is a good question. By current design, we can do in different ways:
1) deleting a blob will remove all chunks associated with the blob
2) deleting the whole database file and restart will remove all chunks.
3) starting nydusd with chunk dedup disabled will not create new chunk dedup.
Ideally, we should support method 1, but haven't implemented yet.

Reply:
OK, may option 1 need a GC mechanism, e.g. long unused blobs should be removed from the database.